lib/src/components/NxAlert/__tests__/NxAlert.test.tsx

```diff
@@ -4,263 +4,158 @@
 
-    expect(contentEl).toExist();
-    expect(contentEl.find('.test-paragraph')).toExist();
-    expect(contentEl.find('.test-container')).toExist();
-  });
+      expect(quickRender({ children }).getByTestId('test-container')).toBeInTheDocument();
+    });
 
-  it('renders the icon passed to it', function() {
-    expect(getNxAlert().find(NxFontAwesomeIcon)).toExist();
-    expect(getNxAlert().find(NxFontAwesomeIcon)).toHaveProp('icon', faBiohazard);
+    it('sets aria-atomic on the top-level element', function() {
+      expect(renderEl()).toHaveAttribute('aria-atomic', 'true');
+    });
 
-    expect(getNxAlert({icon: faCrow}).find(NxFontAwesomeIcon)).toExist();
-    expect(getNxAlert({icon: faCrow}).find(NxFontAwesomeIcon)).toHaveProp('icon', faCrow);
-  });
+    it('passes any other props to the top-level element', function() {
+      const el = renderEl({ id: 'foo', title: 'bar' });
+      expect(el).toHaveAttribute('id', 'foo');
+      expect(el).toHaveAttribute('title', 'bar');
+    });
 
-  it('passes any other props to the div', function() {
-    const component = getNxAlert({ id: 'foo', title: 'baz' });
-    expect(component).toHaveProp('id', 'foo');
-    expect(component).toHaveProp('title', 'baz');
-  });
+    it('renders a Close button if given an onClose prop', async function() {
+      const user = userEvent.setup(),
+          onClose = jest.fn(),
+          view = quickRender({ onClose }),
+          closeBtn = view.getByRole('button', { name: 'Close' });
 
-  it('sets aria-atomic on the div', function() {
-    expect(getNxAlert()).toHaveProp('aria-atomic', true);
-  });
+      expect(closeBtn).toBeInTheDocument();
+      expect(onClose).not.toHaveBeenCalled();
 
-  it('sets the icons\'s aria-label from the iconLabel prop', function() {
-    expect(getNxAlert({ iconLabel: 'foo' }).find(NxFontAwesomeIcon)).toHaveProp('aria-label', 'foo');
-  });
+      await user.click(closeBtn);
+      expect(onClose).toHaveBeenCalled();
+    });
 
-  it('sets the icons\'s aria-hidden to false if the iconLabel is defined', function() {
-    expect(getNxAlert({ iconLabel: 'foo' }).find(NxFontAwesomeIcon)).toHaveProp('aria-hidden', false);
-    expect(getNxAlert().find(NxFontAwesomeIcon)).toHaveProp('aria-hidden', true);
-  });
+    it('does not render a Close button if not given an onClose prop', function() {
+      const view = quickRender(),
+          closeBtn = view.queryByRole('button', { name: 'Close' });
+
+      expect(closeBtn).not.toBeInTheDocument();
+    });
+  }
 
-  it('renders a Close button if given an onClose prop', function() {
-    const onClose = jest.fn(),
-        component = getNxAlert({ onClose });
+  const minimalProps = {
+        children: 'A message to show in an alert',
+        icon: faBiohazard
+      },
+      quickRender = rtlRender(NxAlert, minimalProps);
 
-    expect(component).toContainMatchingElement(NxCloseButton);
-    expect(onClose).not.toHaveBeenCalled();
+  baseTests(NxAlert, minimalProps);
```

Comment:
does the generic NxAlert need to have tests for role assignment as well, or is it considered sufficient when its tested in all the variations of NxAlert?

Reply:
> does the generic NxAlert need to have tests for role assignment as well, or is it considered sufficient when its tested in all the variations of NxAlert?

This made me realize there was room for several improvements around role testing.  All of the different alert variations _can_ receive a role, and the ones that don't provide a role by default should probably have tests proving that.  Addressed in 72da7ca06